### ISSUE TITLE
BigInteger.ToString("x") sometimes prefixed with a 0

### ISSUE THREAD

**Issue by slozier** [dotnet/runtime#43269]
It seems like any time the hex representation would start with an `8`, `9`, ..., `f` the output gets prefixed by a `0`:
```C#
Debug.Assert(new System.Numerics.BigInteger(0x8).ToString("x") == "08");
```
The expected behavior is to get the same result as `int.ToString("x")` - no 0 prefix.

This appears to occur in all versions on .NET (Core, Framework, Mono).

**Comment by am11** [comment]
Modifying https://github.com/dotnet/runtime/blob/f6dc87f43df543bdc761ded7c578ecedc5b39668/src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs#L525 as
```c#
if (head < 0x10 || clearHighF)
```
prints `8` and `9` without leading zeros. However, the comment on line 514 about two's complement suggests that it should retain the high bit, so there might be some catch.

**Comment by vcsjones** [comment]
Right; I think the leading zero is intentional so the string round trips correctly. For example:

```c#
BigInteger bi = BigInteger.Parse("8", System.Globalization.NumberStyles.HexNumber);
Console.WriteLine(bi.ToString());
```

Without the leading zero, this will print "-8".  

We can simulate the behavior of not including the leading zero as such:

```c#
BigInteger a = 8;
BigInteger b = BigInteger.Parse(a.ToString("x").TrimStart('0'), NumberStyles.HexNumber);
Console.WriteLine(a == b); //Prints false
```

`Int32` does not have this behavior, so the leading zero is not required:

```c#
int a = 8;
int b = int.Parse(a.ToString("x").TrimStart('0'), NumberStyles.HexNumber);
Console.WriteLine(a == b); // Prints true
```

**Comment by tannergooding** [comment]
Given that BigInteger isn't implicitly signed or unsigned, I'm not sure this can be made consistent.

You could just say it works with positive/negative hex values (e.g. `0x80` is positive `128` and `-0x80` is `-128`) like C# does (i.e. `(sbyte)(-0x80)` is fine, but `(sbyte)(0x80)` gives CS0221 and needs an unchecked context), but that is a breaking change and would need a compat check.

**Comment by slozier** [comment]
Being able to handle positive/negative hex values would be ideal (basically what I was trying to accomplish). To avoid a break maybe it could be enabled via `NumberStyles.AllowLeadingSign`?

I did find some info about the 0 prefix in the `BigInteger.Parse` docs.

Anyway, closing this out since it can't be made consistent.
